staff/lab/ocf-tv

```diff
@@ -50,91 +41,95 @@ def wait_for_port(host, port, timeout=5, ssh_proc=None):
         if spent > timeout:
             raise TimeoutError('Timed out after {} seconds.'.format(timeout))
 
-def check_volume(val):
-    x = int(val)
-    if abs(x) > 150:
-        raise argparse.ArgumentTypeError('Volume out of bounds: {} not in [0, 150]'.format(val))
+def ssh_to_tv(host_port, remote_port):
+    return subprocess.Popen(['ssh', '-N', '-o', 'ExitOnForwardFailure=yes',
+                             '-o', 'BatchMode=yes',
+                             '-L', '{}:localhost:{}'.format(host_port, remote_port), args.host])
 
-    return val
+def open_vnc(args):
+    port = unused_port()
+    proc = ssh_to_tv(port, 5900)
+    
+    try:
+        wait_for_port('localhost', port, ssh_proc=proc)
+        subprocess.call(['xvncviewer', 'localhost:{}'.format(port)])
+    finally:
+        proc.terminate()
+        proc.wait()
+        
+def tunnel_audio(args = None):
+    port = unused_port()
+    proc = ssh_to_tv(port, 4713)
+
+    try:
+        wait_for_port('localhost', port, ssh_proc=proc)
+        create_audio_tunnel(port)
+        input("Press enter to close the tunnel...")
+    finally:
+        close_audio_tunnel()
+        proc.terminate()
+        proc.wait()
         
+def create_audio_tunnel(local_port = 4710):
+    """Create an explicit PulseAudio tunnel sink to the TV."""
+
+    try:
+        # unload preexisting tunnel sinks if they're there because otherwise
+        # pulse will just create a new one with the name incremented
+        ret = subprocess.call("pactl list sinks " 
+                              "| grep -qE 'Name: {}$'".format(TUNNEL_NAME),
+                              shell=True)
+        if ret is 0:
+            close_audio_tunnel()
+
+        # set up an explicit tunnel sink that points to the local end of our SSH tunnel
+        # if this succeeds, it should return a loaded module index, e.g. '25' that
+        # you can see by running `pactl list modules`
+        ret = subprocess.call(('pactl', 'load-module', 'module-tunnel-sink-new',
+                               'server=tcp:localhost:{}'.format(local_port),
+                               'sink_name=' + TUNNEL_NAME), stdout=subprocess.DEVNULL)
+        
+        ret = subprocess.call(('pactl', 'set-default-sink', TUNNEL_NAME))
+
+        # move everything that's currently playing to the new sink
+        cmd = "pacmd list-sink-inputs \
+              | awk '/index: / { print $2 }' \
+              | xargs -I {} pacmd move-sink-input {} " + TUNNEL_NAME
+                              
+        subprocess.call(cmd, shell=True)
+                              
+    except Exception as e:
+        print(e)
+        exit(1)
```

Comment:
How is this different from not catching? Unless you want to hide the traceback, that is.